client/src/App.tsx

```diff
@@ -83,6 +88,136 @@ const App = () => {
+            return (
+                Math.round(
+                    (parseInt(''+colorArr[0]) * 299 +
+                        parseInt(''+colorArr[1]) * 587 +
+                        parseInt(''+colorArr[2]) * 114) /
+                    1000
+                ) > 150
+            )
+        }
+        if (isSeconds) {
+            fc.fillStyle = color;
+            fc.strokeStyle = color;
+            fc.beginPath();
+            fc.moveTo(FAVICON_SIZE / 2 + sRadius, FAVICON_SIZE / 2)
+            fc.arc(FAVICON_SIZE / 2, FAVICON_SIZE / 2, sRadius, 0, 2 * Math.PI)
+            fc.closePath()
+            fc.fill()
+
+
+            fc.beginPath()
+            fc.moveTo(FAVICON_SIZE / 2, FAVICON_SIZE / 2 - sRadius)
+            // Rounding seconds so when it shows 30 seconds always will show half-way,
+            // even if it's not exactly 30s
+            fc.arc(
+                FAVICON_SIZE / 2,
+                FAVICON_SIZE / 2,
+                sRadius,
+                Math.PI * 1.5,
+                2 * Math.PI * (1 - Math.round(numToShow) / 60) - Math.PI / 2,
+                true
+            )
+            fc.stroke()
+
+
+            fc.fillStyle = isLight(color)? 'black' : 'white';
+            fc.font = `bold ${FAVICON_SIZE * 0.6}px "Roboto", sans-serif`
+            fc.fillText(
+                Math.round(numToShow)
+                    .toString()
+                    .padStart(2, '0'),
+                FAVICON_SIZE / 2,
+                FAVICON_SIZE * 0.575
+            )
+        }
+        else {
+            fc.clearRect(0, 0, FAVICON_SIZE, FAVICON_SIZE)
+
+            fc.fillStyle = color;
+            fc.beginPath()
+            // Rounded square
+            fc.moveTo(0, borderRadius)
+            fc.arc(borderRadius, borderRadius, borderRadius, Math.PI, Math.PI * 1.5)
+            fc.lineTo(FAVICON_SIZE - borderRadius, 0)
+            fc.arc(
+                FAVICON_SIZE - borderRadius,
+                borderRadius,
+                borderRadius,
+                -Math.PI / 2,
+                0
+            )
+            fc.lineTo(FAVICON_SIZE, FAVICON_SIZE - borderRadius)
+            fc.arc(
+                FAVICON_SIZE - borderRadius,
+                FAVICON_SIZE - borderRadius,
+                borderRadius,
+                0,
+                Math.PI / 2
+            )
+            fc.lineTo(borderRadius, FAVICON_SIZE)
+            fc.arc(
+                borderRadius,
+                FAVICON_SIZE - borderRadius,
+                borderRadius,
+                Math.PI / 2,
+                Math.PI
+            )
+            fc.closePath()
+            fc.fill()
+
+            fc.fillStyle = isLight(color)? 'black' : 'white';
+            fc.font = `bold ${FAVICON_SIZE * 0.8}px "Roboto", sans-serif`
+            fc.fillText(''+numToShow, FAVICON_SIZE / 2, FAVICON_SIZE * 0.575)
+        }
+
+        favicon.href = faviconCanvas.toDataURL()
+
+        /** ====== end favicon ====== */
+
```

Comment:
You may want to add a cleanup function to remove the favicon here? ('Tis a `useEffect` thing)

Reply:
Did that in the new commit!